partner_vat_unique/models/res_partner.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2017 Grant Thornton Spain - Ismael Calvo <ismael.calvo@es.gt.com>
+# Copyright 2020 Manuel Calero - Tecnativa
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import config
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    vat = fields.Char(copy=False)
+
+    @api.constrains("vat", "parent_id")
+    def _check_vat_unique(self):
+        for record in self:
+            if record.parent_id or not record.vat:
+                continue
+            test_condition = config["test_enable"] and not self.env.context.get(
```

Comment:
This is not really good practice. Instead, a configuration parameter is better to enable the constraint.

Reply:
Thanks for the feedback! Do you have an example of how to properly implement this with a configuration parameter in Odoo?

Reply:
In ResConfigSettings model:

```
partner_vat_unique = fields.Boolean(
    config_parameter="partner_vat_unique.partner_vat_unique',
    help="Check this if you want to constrain VAT number to be unique"
)
``` 

Then, here, before the loop:

```
if  self.env["ir.config_parameter"]
            .sudo()
            .get_param("partner_vat_unique.partner_vat_unique', False):
```

Reply:
Then, remove all things about test_enable.

And adapt tests accordingly.

Then, put that in a separate commit.

Reply:
I’ve added a commit implementing the configuration parameter for the unique VAT constraint as you suggested. Could you please take a look and let me know if this approach works for you?